rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/cluster.desired_balance/10_basic.yml

```diff
@@ -0,0 +1,136 @@
+---
+setup:
+  - skip:
+      version: " - 8.5.99"
+      reason: "API added in in 8.6.0"
+
+---
+"Test empty desired balance":
+
+  - do:
+      _internal.get_desired_balance: { }
+
+  - gte: { stats.computation_submitted: 0 }
+  - gte: { stats.computation_executed: 0 }
+  - gte: { stats.computation_converged: 0 }
+  - gte: { stats.computation_iterations: 0 }
+  - gte: { stats.computation_converged_index: 0 }
+  - gte: { stats.computation_time_in_millis: 0 }
+  - gte: { stats.reconciliation_time_in_millis: 0 }
+  - match: { routing_table: {} }
+
+---
+"Test get desired balance for single shard":
+  - do:
+      indices.create:
+        index: test
+        body:
+          settings:
+            number_of_shards: 1
+            number_of_replicas: 0
+
+  - do:
+      bulk:
+        index: test
+        refresh: true
+        body:
+          - '{ "index": {"_id": "1"}}'
+          - '{  "field": "foo" }'
+          - '{ "index": {"_id": "2"}}'
+          - '{ "field": "foo 2" }'
+          - '{ "index": {"_id": "3"}}'
+          - '{ "field": "foo 3" }'
+          - '{ "index": {"_id": "4"}}'
+          - '{ "field": "foo 4" }'
+          - '{ "index": {"_id": "5"}}'
+          - '{ "field": "bar" }'
```

Comment:
NIT: this should not affect desired balance

Reply:
:+1: I don't think we need to index any docs in these tests, and would prefer not to do so.

Reply:
Done!